### PR TITLE
moving error handler and sentry plugin to apollo-utils

### PR DESCRIPTION
**Pull request by sri-kirsh**
## Goal

Moving error handler, Sentry plugin to apollo-utils so it can be used as a package across our different services

paired with @kkyeboah 

## Issues:
* https://getpocket.atlassian.net/browse/BACK-1019

## Implementation Decisions
same as this PR: https://github.com/Pocket/client-api/pull/30

We need to update client-api to v3 so we can use `async requestDidStart`

Ticket: https://getpocket.atlassian.net/browse/BACK-1051

### CODE REVIEW

**Review by kschelonka — DISMISSED** [review]
Good call moving the error formatter here as well :) Like the plugin updates!

File: src/plugins/sentryPlugin.ts [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+ * query sent by client)
+ */
+//Copied from https://blog.sentry.io/2020/07/22/handling-graphql-errors-using-sentry
+export const sentryPlugin: ApolloServerPlugin = {
+  requestDidStart(_) {
     /* Within this returned object, define functions that respond
-       to request-specific lifecycle events. */
+                 to request-specific lifecycle events. */
     return {
-      async didEncounterErrors(ctx) {
```
Comment (sri-kirsh) [inline src/plugins/sentryPlugin.ts:10]: todo: revert to earlier async call.

right now, in apollo-utils `async` doesnt work with `ApolloServerPlugin`

Comment (sri-kirsh) [reply src/plugins/sentryPlugin.ts:10]: done

**Review by kkyeboah — APPROVED** [review]
🍐

**Pull request merged**